src/sched/lthread.c

```diff
@@ -461,6 +487,50 @@ void _lthread_free(struct lthread* lt)
     lt = 0;
 }
 
+int __init_utp(void *p, int set_tp)
+{
+	struct lthread_tcb_base *tcb = (struct lthread_tcb_base *)p;
+	tcb->self = p;
+	tcb->schedctx = __scheduler_self();
+	if (libc.user_tls_enabled && set_tp)
+	{
+		if (sgxlkl_enclave->mode == SW_DEBUG_MODE)
+		{
+			int r = __set_thread_area(TP_ADJ(p));
+			if (r < 0)
+			{
+				sgxlkl_fail("Could not set thread area %p: %s\n", p, strerror(errno));
+			}
+		}
+		else
+		{
+			__asm__ volatile("wrgsbase %0" ::"r"(p));
+		}
+	}
+	return 0;
+}
+
+void *__copy_utls(struct lthread *lt, unsigned char *mem, size_t sz)
```

Comment:
Most of this should go away.  Threads created in musl will handle this, so this is just for kernel and host-interface threads.  The only thing that they need is a new random stack canary value.

Reply:
Removing both copy_utls and __init_utp, as they are effectively serving no purpose currently. The only usage in lthread_create calls __init_utp with set_tp = 0, at which point both of these functions are only doing pointer manipulations on local variables.